### PR TITLE
Apply some source-build patches.

### PR DESCRIPTION
**Pull request by crummel**
- Use pinned MSBuild reference version.
- Automatically exclude test projects from source-build.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -18,9 +18,11 @@
...
     <IsTestAssetProject Condition="$(RepoRelativeProjectDir.Contains('testassets'))">true</IsTestAssetProject>
+    <IsOtherTestProject Condition="$(MSBuildProjectName.Contains('IntegrationTesting')) or $(MSBuildProjectName.Contains('TestCommon'))">true</IsOtherTestProject>
     <IsSampleProject Condition="$(RepoRelativeProjectDir.Contains('sample'))">true</IsSampleProject>
     <IsAnalyzersProject Condition="$(MSBuildProjectName.EndsWith('.Analyzers'))">true</IsAnalyzersProject>
     <IsShipping Condition="'$(IsSampleProject)' == 'true' or '$(IsTestAssetProject)' == 'true' or '$(IsBenchmarkProject)' == 'true' or '$(IsUnitTestProject)' == 'true'">false</IsShipping>
+    <ExcludeFromSourceBuild Condition="'$(IsSampleProject)' == 'true' or '$(IsTestAssetProject)' == 'true' or '$(IsBenchmarkProject)' == 'true' or '$(IsUnitTestProject)' == 'true' or '$(IsSpecificationTestProject)' == 'true' or '$(IsOtherTestProject)' == 'true'">true</ExcludeFromSourceBuild>
```
Comment (dougbu) [inline Directory.Build.props]: Changes to this file should instead be in Directory.Build.targets. See https://github.com/aspnet/AspNetCore/blob/df1477b13f8bc256f5001ec5f7dd70936394c366/Directory.Build.targets#L14

File: eng/tools/RepoTasks/RepoTasks.csproj [hunk 2]
```diff
@@ -17,9 +17,9 @@
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == '$(DefaultNetCoreTargetFramework)'">
-    <PackageReference Include="Microsoft.Build.Framework" Version="15.8.166" />
-    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="15.8.166" />
-    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="15.8.166" />
+    <PackageReference Include="Microsoft.Build.Framework" Version="15.7.179" />
+    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="15.7.179" />
+    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="15.7.179" />
```
Comment (dougbu) [inline eng/tools/RepoTasks/RepoTasks.csproj]: Why do these versions need to be _reduced_?

**Comment by crummel** [comment]
Nope, this has been fixed.  Sorry for the trouble, closing.

**Pull request closed without merging**